content/Software Support/PLC IDE/If-PLC-IDE-isnt-working-with-your-device.md

```diff
@@ -0,0 +1,229 @@
+
+### If PLC IDE becomes unresponsive when connecting to the device
+
+PLC IDE may become unresponsive if there isn't a response on the targeted port.
+
+Follow these steps:
+
+1. Ensure the correct port is configured for the Modbus protocol:
+   1. Select **On-line > Set up communication**.
+   1. Click on **Properties** to open the Modbus configuration.
+   1. Under Communication, ensure the right **Port** is selected.
+   1. See the [If you can't find the serial port for your device](#if-you-cant-find-the-serial-port-for-your-device) section for help finding the right port.
+1. Close other applications that may be using the port, such as the Serial Monitor in Arduino IDE.
+1. Restart your computer.
+
+---
+
+## If you see an error when programming the device
+
+When there is an issue installing the runtime or downloading your code to the device, error messages may appear in the **Resources** tab of the **Output** console.
+
+In this section:
+
+- [Error: "Cannot download sketch file error code: 1"](#error-cannot-download-sketch-file-error-code-1)
+- [Cannot generate profile file error message: Error: Path not found](#cannot-generate-profile-file-error-message-error-path-not-found)
+- [Error: invalid path creating config dir](#error-invalid-path-creating-config-dir)
+- [“Error” in the bottom right corner on first use of the board](#error-in-the-bottom-right-corner-on-first-use-of-the-board)
+
+> [!NOTE]
+> If you see a **dfu-util** warning or error that's not in this list, see [dfu-util errors when uploading (exit status 74)](https://support.arduino.cc/hc/en-us/articles/11011849739804-dfu-util-errors-when-uploading-exit-status-74).
+
+### Error pop-up: "Cannot download sketch file (error code: 1)"
+
+This error may occur when attempting to download the runtime sketch.
+
+Solutions:
+
+1. [Verify that the correct port is selected in the manual sketch download widget.](https://support.arduino.cc/hc/en-us/articles/16724283965596)
+
+1. Double-tap the device’s reset button to put it in bootloader mode. First, try selecting the correct COM port. If that doesn’t work, set the Port to “Automatic (BootLoader)” and attempt to download the sketch again.
+
+1. Close any applications blocking the port, such as the Arduino IDE serial monitor.
+
+1. [Reformat the QSPI flash memory](https://support.arduino.cc/hc/en-us/articles/16206977438748-Reset-the-flash-memory-on-STM32H747-based-devices)
+
+1. Create a new project and attempt to download the runtime again.
+
+### Cannot generate profile file (error message: Error: Path not found)
+
+If you see this error, it means that the `sketch.yaml` file is missing from `C:\Users\USERNAME\AppData\Local\T\sketch.yaml`.
+
+When installing PLC IDE, a **User Account Control** prompt will ask for administrator privileges. Some files that are used by PLC IDE will be created in the `C:\Users\ADMIN-USERNAME\AppData\Local\T` directory for that admin account.
```

Comment:
@marqdevx We try to make use of environment variables where we can. This situation is a bit complicated since we are dealing with two different locations (for the admin and for the regular user.

Later on, we may want to consider a dedicated article for setting up the "T" folder, where we can also include some more specific command examples.